models/resource.go

```diff
@@ -0,0 +1,18 @@
+package models
+
+type kvmap map[string]string
+
+// Metadata represents the metadata section of a Kubernetes resource.
+type Metadata struct {
```

Comment:
a stylistic nit pick (feel free to disagree here) but I prefer it if the parent struct goes first, then children, so reorder it like: 
```golang
Resource struct { 
    ...
    Metadata Metadata
} 
// then define metadata struct

Metadata struct { 
    ....etc.
} 
```
it's easier to read imo, especially if the models get dense

Reply:
I'm conditioned by decades of languages that fail if you try to refer to a data type before it's been defined. I mean, that fails in C; the following gives you `main.c:2:5: error: unknown type name ‘bar’`:

```
typedef struct {
    bar *Bar;
} foo;

typedef struct {
    int id;
} bar;
```

And it fails in Python, which gives you `NameError: name 'bar' is not defined` for this:

```
class foo(bar):
    def __init__(self):
        self.bar = bar()


class bar:
    ...
```

Reply:
I get that, if you feel strongly on this I won't push further, but it does seem to me that the standard for declaring struct representation of yaml/jsons seems to follow parent -> child type ordering, e.g. see how types are declared for argocd [here](https://github.com/argoproj/argo-cd/blob/master/pkg/apis/application/v1alpha1/types.go)